src/mumble/API.h

```diff
@@ -0,0 +1,188 @@
+							   api_promise_t *promise);
+	void requestLocalUserTransmissionMode_v_1_0_x(mumble_plugin_id_t callerID,
+												  mumble_transmission_mode_t transmissionMode, api_promise_t *promise);
+	void getUserComment_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, mumble_userid_t userID,
+								const char **comment, api_promise_t *promise);
+	void getChannelDescription_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection,
+									   mumble_channelid_t channelID, const char **description, api_promise_t *promise);
+	void requestUserMove_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, mumble_userid_t userID,
+								 mumble_channelid_t channelID, const char *password, api_promise_t *promise);
+	void requestMicrophoneActivationOverwrite_v_1_0_x(mumble_plugin_id_t callerID, bool activate,
+													  api_promise_t *promise);
+	void requestLocalMute_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, mumble_userid_t userID,
+								  bool muted, api_promise_t *promise);
+	void requestLocalUserMute_v_1_0_x(mumble_plugin_id_t callerID, bool muted, api_promise_t *promise);
+	void requestLocalUserDeaf_v_1_0_x(mumble_plugin_id_t callerID, bool deafened, api_promise_t *promise);
+	void requestSetLocalUserComment_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection,
+											const char *comment, api_promise_t *promise);
+	void findUserByName_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, const char *userName,
+								mumble_userid_t *userID, api_promise_t *promise);
+	void findChannelByName_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, const char *channelName,
+								   mumble_channelid_t *channelID, api_promise_t *promise);
+	void getMumbleSetting_bool_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, bool *outValue,
+									   api_promise_t *promise);
+	void getMumbleSetting_int_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, int *outValue,
+									  api_promise_t *promise);
+	void getMumbleSetting_double_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, double *outValue,
+										 api_promise_t *promise);
+	void getMumbleSetting_string_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, const char **outValue,
+										 api_promise_t *promise);
+	void setMumbleSetting_bool_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, bool value,
+									   api_promise_t *promise);
+	void setMumbleSetting_int_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, int value,
+									  api_promise_t *promise);
+	void setMumbleSetting_double_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, double value,
+										 api_promise_t *promise);
+	void setMumbleSetting_string_v_1_0_x(mumble_plugin_id_t callerID, mumble_settings_key_t key, const char *value,
+										 api_promise_t *promise);
```

Comment:
I'm not sure about this, the data types are pretty limited.

What about a single function that takes a pointer to the value and its size?

Reply:
That way you'd have to use `reinterpret_cast` a lot which in general seems to be undesired.

The data types covered so far seem to fit all settings that we currently have. Do you know of some that can't be accessed using these types?
And even if it turns out that we need more types in the future, we can simply add new functions for that new type.

Reply:
I was actually mostly concerned about memory usage.

Considering `bool` is 1 byte big, I would provide functions for the following datatypes:
```c
int8_t
int16_t
int32_t

float

char *
```

This should give enough flexibility in terms of memory usage optimization and available storage.

Reply:
I really don't think we have to care about memory efficiency here. I expect these functions to be used only very infrequently and a few bytes give or take should really not make any difference whatsoever.

Thus providing types for the different data type sizes just makes it even harder to figure out the correct type to use. Plus if at some point we decide that a given setting should be a `double` instead of a `float`, then we could not do that without breaking backwards compatibility for older plugins.
Therefore I think defaulting to the bigger type should be fine.

Nonetheless I think we should probably change `int` to `int64_t` in order to not make that parameter platform-dependent.

Reply:
Okay, in that case:

```c
int64_t

double

char *
```

Even if a plugin developer wants to take a memory address in input there should be no issues whatsoever because it's not absolute. In fact, there are cases where offsets can even fit in a 16 bit variable (`0x0000` - `0xFFFF`).

Reply:
The only real difference seems to be that the `bool` type would be missing. What would be the benefit of that?
I think I still don't really understand your argumentation here :eyes:

Reply:
Yeah, sorry, add `bool` to the list.

Reply:
But then this is exactly what we have right now, isn't it? So is there still any need for a change? Given that you marked this discussion as resolved, I guess not? :thinking:

Reply:
Correct.